plugins/org.eclipse.fordiac.ide.export.forte_ng/src/org/eclipse/fordiac/ide/export/forte_ng/simple/SimpleFBImplTemplate.xtend

```diff
@@ -35,12 +36,9 @@ class SimpleFBImplTemplate extends BaseFBImplTemplate<SimpleFBType> {
 	override generateExecuteEvent() '''
 		void «FBClassName»::executeEvent(const TEventID paEIID, CEventChainExecutionThread *const paECET) {
 		  switch(paEIID) {
-		    «FOR event : type.interfaceList.eventInputs»
-		    	case «event.generateEventID»:
-		    	  «FOR alg : type.algorithm.filter[name == event.name]»
-		    	  	«alg.generateAlgorithmName»();
-		    	  «ENDFOR»
-		    	  break;
+		    «FOR state : type.simpleECStates»
+		     	case «state.inputEvent.generateEventID»:
+		     	  enterState«state.name»(paECET)
```

Comment:
Please add the missing semicolon at the end of this line and also a `break;` afterwards.

Reply:
thanks for catching that